src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -165,17 +165,11 @@ public function normalize(mixed $data, string $format = null, array $context = [
         }
 
         if (is_iterable($data)) {
-            if (is_countable($data) && 0 === \count($data)) {
-                switch (true) {
-                    case ($context[AbstractObjectNormalizer::PRESERVE_EMPTY_OBJECTS] ?? false) && \is_object($data):
-                        if (!$data instanceof \ArrayObject) {
-                            trigger_deprecation('symfony/serializer', '5.4', 'Returning empty object of class "%s" from "%s()" is deprecated. This class should extend "ArrayObject".', get_debug_type($data), __METHOD__);
-                        }
-
-                        return $data;
-                    case ($context[self::EMPTY_ARRAY_AS_OBJECT] ?? false) && \is_array($data):
-                        return new \ArrayObject();
-                }
+            if (is_countable($data) && 0 === \count($data) && (
+                ($context[AbstractObjectNormalizer::PRESERVE_EMPTY_OBJECTS] ?? false) && \is_object($data) ||
+                ($context[self::EMPTY_ARRAY_AS_OBJECT] ?? false) && \is_array($data)
```

Comment:
don't we miss bracket around both operands of `||`?

Reply:
I don't think so, why ? (There are present and tests are OK)

Reply:
I read this as `A && B || C && D`, and I don't know if this is the same as `(A && B) || (C && D)` (which is what we want, isn't it?)

Reply:
`A && B || C && D` and `(A && B) || (C && D)` are exactly the same. I don't get it 🤔